src/library/scala/collection/immutable/HashMap.scala

```diff
@@ -654,6 +879,39 @@ private final class BitmapIndexedMapNode[K, +V](
     new BitmapIndexedMapNode[K, V](dataMap ^ bitpos, nodeMap, dst, dstHashes, size - 1, cachedJavaKeySetHashCode - keyHash)
   }
 
+  /** Variant of `copyAndMigrateFromInlineToNode` which mutates `this` rather than returning a new node.
+    *
+    * @param bitpos the bit position of the data to migrate to node
+    * @param keyHash the improved hash of the key currently at `bitpos`
+    * @param node the node to place at `bitpos`
+    */
+  def migrateFromInlineToNodeInPlace[V1 >: V](bitpos: Int, keyHash: Int, node: MapNode[K, V1]): this.type = {
```

Comment:
This functionality is now duplicated here and in the builder, probably best to use this method in the builder and remove that version.